src/RamTransforms.cpp

```diff
@@ -565,18 +565,18 @@ bool HoistAggregateTransformer::hoistAggregate(RamProgram& program) {
     // most outer permissible aggregate is hoisted
     visitDepthFirst(program, [&](const RamQuery& query) {
         // aggregate description
-        int aggLevel = -1;
-        AggregateFunction newFun;
-        const RamRelation* newRef;
-        RamExpression* newExp;
-        RamCondition* newCond;
+        int aggLevel;
+        AggregateFunction aggFun;
```

Comment:
Could all this be simplified by capturing the removed aggregate in a single variable instead of its internal state?

Reply:
Daniel was struggling with this because it is not an easy problem in the apply semantics of RamNodes. 

The issue is that we would need the original aggregate and divert its nested operation to the operation at the point at where it is inserted. 

Daniel's solution was to describe the hoistable aggregate and create a new instance rather than reusing the old aggregrate instance (which is thrown away by the smart-pointer semantics). 

Perhaps we accept the pull-request and make it an SW-Eng issue since we have an urgent functional problem at hand.